### ISSUE TITLE
Package compatibility issue on macOS ARM64 (M3 Pro)

### ISSUE THREAD

**Issue by sanjeevpenupala** [yisol/IDM-VTON#70]
# Description
Unable to create the environment on macOS with ARM64 architecture (M3 Pro chip) due to package compatibility issues.

# Body
When attempting to create the environment using the provided environment.yaml file on a macOS system with an ARM64 architecture (M3 Pro chip), the following error occurs:
```
> conda env create -f environment.yaml
Channels:
 - pytorch
 - nvidia
 - defaults
Platform: osx-arm64
Collecting package metadata (repodata.json): done
Solving environment: failed

PackagesNotFoundError: The following packages are not available from current channels:

  - torchvision==0.15.2=py310_cu118
  - torchtriton==2.0.0=py310
  - torchaudio==2.0.2=py310_cu118
  - pytorch-cuda==11.8=h7e8668a_5
  - pytorch==2.0.1=py3.10_cuda11.8_cudnn8.7.0_0
  - python==3.10.0=h12debd9_5
  - pip==23.3.1=py310h06a4308_0

Current channels:

  - https://conda.anaconda.org/pytorch/osx-arm64
  - https://conda.anaconda.org/nvidia/osx-arm64
  - https://repo.anaconda.com/pkgs/main/osx-arm64
  - https://repo.anaconda.com/pkgs/r/osx-arm64

To search for alternate channels that may provide the conda package you're
looking for, navigate to

    https://anaconda.org

and use the search bar at the top of the page.
```

The error indicates that the required packages are not available for the osx-arm64 platform in the current channels.

# Possible Solutions

- Check for ARM64-compatible versions of the required packages and update the environment.yaml file accordingly.
- Provide an alternative environment.yaml file specifically for macOS ARM64 systems, including the M3 Pro chip.
- Include instructions or workarounds for running the code on ARM64 systems, such as using Docker containers or virtual machines.

# System Information

- Operating System: macOS Sonoma 14.1
- CPU Architecture: ARM64 (Apple M3 Pro chip)

**Comment by diamsa** [comment]
Hi, i just was trying to run the conda environment and i was getting the same error on my M2. Did you modify the ```environment.yaml``` to get it working?

**Comment by spenpal** [comment]
Yes, I did modify the `environment.yaml` to make the packages Mac compatible. However, the main issue that arises is that all the code is written for NVIDIA CUDA. So, you would have to modify the code base to make it compatible with MPS.

Which is too much of a hassle, so I abandoned attempting to run it locally.

**Comment by diamsa** [comment]
That's what i concluded as well. I'll explore the possibility of running it with Docker.

I let you know if i get it working

**Comment by spenpal** [comment]
Sounds good. Keep me posted!

**Comment by FurkanGozukara** [comment]
you can run on cloud i have a full tutorial

### [**IDM-VTON: The Most Amazing Virtual Clothing Try On Application - RunPod - Massed Compute - Kaggle**](https://youtu.be/LeHfgq_lAXU)

[![image](https://cdn-uploads.huggingface.co/production/uploads/6345bd89fe134dfd7a0dba40/G3um_XDD1Pg2n_wQ1qwT2.png)](https://youtu.be/LeHfgq_lAXU)

**Comment by emircetinmemis** [comment]
> That's what i concluded as well. I'll explore the possibility of running it with Docker.
> 
> I let you know if i get it working

Hi there, did you achieve this ?

**Comment by diamsa** [comment]
Hi, I didn’t find time to set up a docker version :(
